yajsapi/executor/index.ts

```diff
@@ -644,7 +637,7 @@ export class Executor {
 
       emit(new events.WorkerStarted({ agr_id: agreement.id() }));
 
-      if (self._worker_cancellation_token.cancelled) { return; }
+      if (self.state && self.state.worker_cancellation_token.cancelled) { return; }
```

Comment:
I'd add some kind of type casting, to tell the compiler that `self.state` is expected to be defined at this point. Otherwise it may be a bit confusing for the reader.

Reply:
The same applies to several other places when `self.state.worker_cancellation_token` appears

Reply:
Fixed